### PR TITLE
 feat(app/*) Integrate MailChimp with IBC.

### PR DESCRIPTION
**Pull request by cjrowden**

### CODE REVIEW

**Comment by jspizziri** [comment]
@mrogers-5s & @cjrowden it might be better to push this logic down into the model's rather than putting it in the controllers. Is there a reason we didn't do that?

**Comment by mrogers-5s** [comment]
@jspizziri When I added the abandoned cart tracking logic for SJC, I originally moved the MailChimp sync into the models so it was only defined in one place. You asked me to move it back into the controllers.

I believe that was to avoid sending MailChimp requests when changes were saved in Spree to fields that aren't stored in MailChimp. (And also so in-progress admin orders didn't trigger abandoned cart emails.

I suggested to Christopher that we stick with the same approach for users, but maybe it does make sense to move the user logic down into the models.

File: app/models/spree/line_item_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Spree::LineItem.class_eval do
+
+  after_save :notify_mail_chimp
+
+  def notify_mail_chimp
```
[the remark this answers is not in the document]
Comment (mrogers-5s) [reply app/models/spree/line_item_decorator.rb]: @jspizziri This is the decorator for the line item model.

[the remark this answers is not in the document]
Comment (jspizziri) [reply app/models/spree/line_item_decorator.rb]: @mrogers-5s wow... thanks for that. The old eye's don't work as well as they used to 😜

**Pull request merged**